Voicemail: Add option to lock the password (i.e. for shared mailboxes)

hannes427 opened the pull request:
If the password is prefix by '-', the password is locked (see https://github.com/asterisk/asterisk/blob/2d7948fa483c8703cf9948811fd67147f4d560d4/configs/samples/voicemail.conf.sample#L330 for details).

Bugfix FreePBX/issue-tracker#151

hannes427 committed:
Voicemail: Add option to lock the password (i.e. for shared mailboxes)

If the password is prefix by '-', the password is locked (see
https://github.com/asterisk/asterisk/blob/2d7948fa483c8703cf9948811fd67147f4d560d4/configs/samples/voicemail.conf.sample#L330
for details).

Bugfix FreePBX/issue-tracker#151

hannes427 commented:
@kguptasangoma This is the new pull request without the i18n files.

miken32 commented:
If this is intended as a security feature, it's very trivial to just remove the dash from the hidden input value. The server-side code should be checking for a leading dash and not allowing updates from UCP in that case.

miken32 reviewed (commented):
`str_starts_with($settings['pwd'], '-')` seems simpler to read at a glance than the regex, IMO

hannes427 committed:
Merge branch 'FreePBX:release/17.0' into Issue-151-new

hannes427 committed:
Voicemail: Add option to lock the password (i.e. for shared mailboxes)

If the password is prefixed by '-', the password is locked (see https://github.com/asterisk/asterisk/blob/2d7948fa483c8703cf9948811fd67147f4d560d4/configs/samples/voicemail.conf.sample#L330 for details).

* Add server-side check in ucp/Voicemail.class.php check for leading '-'
  in the old password before password is saved
* Replaced regex in ucp/views/vmsettings.php by str_starts_with()
* Modified regex in functions.inc.php to allow voicemail passwords with leading '-' in admin-panel

Bugfix FreePBX/issue-tracker#151

hannes427 committed:
Voicemail: Add option to lock the password (i.e. for shared mailboxes)

If the password is prefixed by '-', the password is locked (see https://github.com/asterisk/asterisk/blob/2d7948fa483c8703cf9948811fd67147f4d560d4/configs/samples/voicemail.conf.sample#L330 for details).

* Add server-side check in ucp/Voicemail.class.php check for leading '-'
  in the old password before password is saved
* Replaced regex in ucp/views/vmsettings.php by str_starts_with()
* Modified regex in functions.inc.php to allow voicemail passwords with leading '-' in admin-panel

hannes427 committed:
Voicemail: Add option to lock the password (i.e. for shared mailboxes)

If the password is prefixed by '-', the password is locked (see https://github.com/asterisk/asterisk/blob/2d7948fa483c8703cf9948811fd67147f4d560d4/configs/samples/voicemail.conf.sample#L330 for details).

* Add server-side check in ucp/Voicemail.class.php check for leading '-'
  in the old password before password is saved
* Replaced regex in ucp/views/vmsettings.php by str_starts_with()
* Modified regex in functions.inc.php to allow voicemail passwords with leading '-' in admin-panel

Bugfix FreePBX/issue-tracker#151

hannes427 commented:
@miken32 Thank you for your comments. Added a new PR with your modifications

hannes427 committed:
Add option to lock the password (i.e. for shared mailboxes)

* Removed displaying '-' from Voicemail Pin in UCP if password is
  locked

kapilgupta01 commented:
hi @miken32 thanks for your feedback on this PR. Just want to know how does this PR now looks to you?

miken32 commented:
Much improved
